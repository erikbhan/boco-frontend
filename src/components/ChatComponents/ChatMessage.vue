<template>
  <div v-bind:class="'blob-container ' + this.side()">
    <div v-bind:class="this.color() + ' message-container ' + this.textColor()">
      <span class="message">{{ this.message.content }}</span>
      <span class="">{{ this.calculateTime() }}</span>
    </div>
  </div>
</template>

<style scoped>
.blob-container {
  display: flex;
  max-width: 100%;
}

.message {
  word-break: break-word;
  display: block;
}

.message-container {
  border-radius: 10px;
  max-width: 70%;
  padding: 0.75rem;
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
}
</style>
<script>
import { parseCurrentUser } from "@/utils/token-utils";
//block text-xs text-right
export default {
  props: {
    message: Object,
  },
  data() {
    return {
      abc: "abcde",
    };
  },
  computed: {
    userID() {
      return parseCurrentUser().accountId;
    },
  },
  methods: {
    color() {
      //console.log(this.userID);
      return this?.message.from == this.userID
        ? "bg-gray-300"
        : "bg-primary-medium";
    },
    textColor() {
      return this?.message.from == this.userID ? "text-gray-900" : "text-white";
    },
    side() {
      return this?.message.from == this.userID
        ? "justify-start"
        : "justify-end";
    },
    calculateTime() {
      //let time = this.message.from;
      // Calculate time when message was sent
      let date = new Date(this.message.timestamp);
      let hours = date.getHours();
      let minutes = "0" + date.getMinutes();
      let formattedTime = hours + ":" + minutes.substr(-2);
      return formattedTime;
    },
  },
};
</script>
