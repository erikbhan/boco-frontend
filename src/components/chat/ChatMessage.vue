<template>
    <div v-bind:class="'w-full break-words flex ' + side()">
        <div style="max-width: 70%" v-bind:class="this.color() + ' rounded px-5 py-2 my-2 relative ' + this.textColor() ">
            <span class="block">{{this.message.content}} {{this.message.from}}</span>
            <span class="block text-xs text-right">{{this.calculateTime()}}</span>
        </div>
    </div>
</template>

<script>
import {parseCurrentUser} from "@/utils/token-utils";

export default {
    props: {
        message: Object
    },
    data() {
        return {
            abc: "abcde"
        }
    },
    computed: {
        userID() {
            return parseCurrentUser().accountId;
        }
    },
    methods: {
        color() { 
            console.log(this.userID);
            return  this?.message.from == this.userID  ? "bg-gray-300" : "bg-blue-600";
        },
        textColor() {
            return  this?.message.from == this.userID ? "text-gray-900" : "text-white";
        },
        side () {
            return this?.message.from == this.userID ? "justify-start" : "justify-end"
        },
        calculateTime() {
            //let time = this.message.from;
            // Calculate time when message was sent
            let date = new Date(Date.now());
            let hours = date.getHours();
            let minutes = "0" + date.getMinutes();
            let formattedTime = hours + ':' + minutes.substr(-2);
            return formattedTime;
        }
    }
}
</script>