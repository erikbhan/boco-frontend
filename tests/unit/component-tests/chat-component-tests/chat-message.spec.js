import { mount } from "@vue/test-utils";
import ChatMessage from "@/components/ChatComponents/ChatMessage.vue"

describe("ChatMessageComponent", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(ChatMessage, {
        //passing prop to component
        computed: {
          userID() {
            return 1;
          }
        },
        props: {
            message: {
              id: 1,
              content: "string",
              timestamp: "2022-04-29T08:20:42.850Z",
              from: 1,
              to: 3
            } 
        }
    });
  });

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("is instantiated", () => {
    expect(wrapper.exists()).toBeTruthy();
  });
});