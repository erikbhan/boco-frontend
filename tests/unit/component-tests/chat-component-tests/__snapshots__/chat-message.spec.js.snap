// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatMessageComponent renders correctly 1`] = `
<div
  class="w-full break-words flex justify-start"
>
  <div
    class="bg-gray-300 rounded px-5 py-2 my-2 relative text-gray-900"
    style="max-width: 70%;"
  >
    <span
      class="block"
    >
      string 1
    </span>
    <span
      class="block text-xs text-right"
    >
      10:20
    </span>
  </div>
</div>
`;
